server.go

```diff
@@ -4099,9 +4210,51 @@ func (s *server) OutboundPeerConnected(connReq *connmgr.ConnReq, conn net.Conn)
 		s.removePeer(connectedPeer)
 		s.ignorePeerTermination[connectedPeer] = struct{}{}
 		s.scheduledPeerConnection[pubStr] = func() {
-			s.peerConnected(conn, connReq, false)
+			s.peerConnected(conn, connReq, false, access)
+		}
+	}
+}
+
+// assignPeerPerms assigns a new peer its permissions.
+func (s *server) assignPeerPerms(remotePub *btcec.PublicKey) (peerAccessStatus,
+	error) {
+
+	var access peerAccessStatus
+
+	shouldDisconnect, err := s.authGossiper.ShouldDisconnect(remotePub)
+	if err != nil {
+		return access, err
+	}
+
+	if shouldDisconnect {
+		return access, ErrGossiperBan
```

Comment:
nit: I think it'd be better to explicitly return `peerStatusRestricted` here and above.

Reply:
imo doesn't matter, we're returning an error

Reply:
on second thought, we're dealing with access control so I'll explicitly return